### PR TITLE
Upgrade to pyproject.toml

### LINKED ISSUE #104: Improve package management: switching to `pyproject.toml` from `setup.py`
## Use Case

_Please provide a use case to help us understand your request in context_
Currently, we use `setup.py` for managing package configurations. We want to transition to `pyproject.toml` as it supports better dependency resolution with more python tools, and will improve integration with our js build tools.

### PR DESCRIPTION
**Pull request by rugeli**
Problem
=======
What is the problem this work solves, including
closes #104 

Estimate review time: 5mins (tiny)

Solution
========
What I/we did to solve this problem
- updated all `questionary` `ask` functions that were previously missed to `unsafe_ask`  
- removed `setup.py`
- added `pyproject.toml`
- tested with new environment 

Todo next: 
- add more necessary project metadata
- decide on a license 

## Type of change
Please delete options that are not relevant.

* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. Follow the [instruction](https://github.com/allen-cell-animated/cell-feature-data/tree/feature/python-script?tab=readme-ov-file#option-1-use-the-python_dataset_creator-package) in README
2. This change shouldn't affect any existing functionalities. Running `create-dataset` should work the same

### CODE REVIEW

File: cell_feature_data/bin/create_dataset.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def create_single_dataset(output_path: str, for_megaset: bool = False):
     file_path = questionary.path(
         "Enter a valid file path:",
         validate=lambda text: True if len(text) > 0 else "File path cannot be empty.",
-    ).ask()
```
Comment (ShrimpCryptid) [inline cell_feature_data/bin/create_dataset.py:54]: Why did you need to swap for `unsafe_ask`? Do we just want to ignore keyboard interrupt commands (e.g. Ctrl + C)

Comment (rugeli) [reply cell_feature_data/bin/create_dataset.py:54]: By using unsafe_ask, the keyboard interrupts(Ctrl+C) won't be caught by each individual question. Instead, we can catch and handle these interrupts in one place as exceptions. I have tested each unsafe_ask and they worked as expected.

More details in Questionary [documentation ](https://github.com/allen-cell-animated/cell-feature-data/pull/110)

**Review by ShrimpCryptid — APPROVED** [review]
Looks good!

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+readme = "README.md"
+requires-python = ">=3.8"
+classifiers = [
+"Development Status :: 4 - Beta",
+"Natural Language :: English",
+"Programming Language :: Python :: 3.7",
+"Programming Language :: Python :: 3.8",
+"Programming Language :: Python :: 3.9",
+"Programming Language :: Python :: 3.10",
+"Programming Language :: Python :: 3.11",
```
Comment (toloudis) [inline pyproject.toml:19]: I suggest adding 3.12 to this list and removing 3.7. (line 11 already says it requires Python >= 3.8)   Even Python 3.8 is nearing its end-of-life: https://devguide.python.org/versions/

**Review by toloudis — APPROVED** [review]
Looks good! One minor quick thing to update

**Review by toloudis — APPROVED** [review]
